### PR TITLE
Environment changes

### PR DESCRIPTION
**Pull request by dopplershift**
- Move to using miniconda
- Use a separate conda environment rather than root
- Put all of the dependencies into a single `environment.yml` with the `conda-forge` channel enabled.

### CODE REVIEW

**Comment by dopplershift** [comment]
from @scollis via email:
>Ok..

>Need this change:


```bash
$conda_bin_path'conda' env create -f /home/ubuntu/AMS_radar_in_the_cloud/ec2_setup/environment.yml
source $conda_bin_path'activate' ams-workshop
```
>Now there is another issue.. for some reason the shell script does not let me source or run files.. 
>for instance the line 
```bash
. .bashrc 
```
>has no impact

>and, more problematic, it seems 

```
source $conda_bin_path'activate' ams-workshop
```

>has not effect..

**Comment by scollis** [comment]
Also seeing issue with no numpy found.
-EDIT: This is due to the env not being activated

**Comment by scollis** [comment]
Yep.. definite issues with sourcing and changing environment variables in the shell script.. perhaps we need to change it into several scripts..

**Comment by jjhelmus** [comment]
Why not use export the change to the PATH inside the script as well as writing it to `.bashrc`

**Comment by dopplershift** [comment]
I was working on that as well :grin:

File: ec2_setup/setup_ec2.sh [hunk 1]
```diff
@@ -10,46 +10,37 @@ echo '--------'
...
 
-conda_bin_path="/home/ubuntu/anaconda3/bin/" 
+conda_bin_path="/home/ubuntu/miniconda3/bin/"
 
 echo "APPENDING PATH"
 cd ~
-echo 'PATH="/home/ubuntu/anaconda3/bin:$PATH"' >> .bashrc
+echo 'PATH="/home/ubuntu/miniconda3/bin:$PATH"' >> .bashrc
 
 . .bashrc
```
Comment (jjhelmus) [inline ec2_setup/setup_ec2.sh]: How about: `export PATH="$HOME/miniconda3/bin:$PATH"`

**Comment by scollis** [comment]
This is the lesser of the two issues.. the greater is how to make the source activate take effect?

**Comment by jjhelmus** [comment]
> source activate take effect

In the script or in the calling shell?  The former works fine if `activate` is on the path.  The later requires that that `setup_ec2.sh` be sourced not executed.

**Comment by dopplershift** [comment]
I think @scollis means in the script. It doesn't seem to be working, since the install of pyart can't find numpy, even here on osx....

**Comment by scollis** [comment]
I am trying a few methods.. I have split the script into three coming back into the top script when a source is called.. This works for the path appending (source .bashrc)
but NOT source activate.. Exactly @dopplershift 

Some googling sees others with the same problem but no stackoverflow green ticks..

**Comment by scollis** [comment]
I may ping the anaconda list

**Comment by jjhelmus** [comment]
Py-ART finds numpy fine in my setup.  Let me clean it up a bit so I can share.

**Comment by scollis** [comment]
Great. Thanks @jjhelmus

**Comment by jjhelmus** [comment]
dopplershift/AMS_radar_in_the_cloud#1

**Comment by dopplershift** [comment]
@scollis how about now?

**Comment by scollis** [comment]
Just slaying another dragon.. I will test soon.

**Comment by scollis** [comment]
So there is an issue here:

https://github.com/dopplershift/AMS_radar_in_the_cloud/blob/conda-forge/ec2_setup/setup_ec2.sh#L36

on line 26 there is cd ~ so the path is not correct

**Comment by dopplershift** [comment]
FYI, you can go to the "Files Changed" tab above and click on the "+" icon next to the line you want to note and call it out in this PR.

Will look at that.

**Comment by jjhelmus** [comment]
Do we need to serve this over https?  With can remove all the logic in generating the certs, not have to help users bypass their browsers security, etc if we are alright with password protected http.

**Comment by scollis** [comment]
I'm cool with this.. I just boilerplate-ed it from the example

**Comment by dopplershift** [comment]
I'm not *super* comfortable with plaintext passwords. But as long as we make it clear that's the case, and that this is ok because we're using a throw-away instance, it might be ok.

Counter-point: We're showing them the way that, in no uncertain terms, they have no business running outside the workshop.

**Comment by scollis** [comment]
Perhaps have two .sh files _secure and _unsecure

**Comment by jjhelmus** [comment]
> not super comfortable with plaintext passwords

Agree, but I think encouraging them to click through browser cert exceptions might be worse.  Plus, I don't want to try figuring out how to do that on IE/Edge which I'm sure someone will have.

**Comment by dopplershift** [comment]
As long as we indicate the poor security and that they shouldn't proceed this way as a production use of the notebook, I'm ok.

**Comment by jjhelmus** [comment]
Alternatively, we can provide defaults for the cert so that users are not asked those questions and include a `start_notebook.sh` script.  Hosting will be over HTTPS with an unrecognized certificate but would be non-plaintext.

I can have a PR for either option in a few minutes.

**Pull request merged**